ChangeLog.d/add_fallthrough_macro.txt

```diff
@@ -0,0 +1,3 @@
+Features
+   * Add a macro to mark deliberate fallthrough in a case statement, to stop
+   warnings being generated with newer compilers.
```

Comment:
It's not “newer compilers”, it's Clang specifically. I wouldn't rate this as a “feature”. In the past, we've classified silencing compiler warnings as either “Bugfix” (the bug being that the build fails) or “Changes”.
```suggestion
Changes
      * Fix the build with Clang 10 with -Wimplicit-fallthrough when MBEDTLS_USE_PSA_CRYPTO is enabled.
```